src/ledger/LedgerStateTests.cpp

```diff
@@ -0,0 +1,655 @@
+        ls2.commit();
+    }
+    {
+        LedgerState ls(root);
+        dbInflationWinners = ls.loadInflationWinners(10, 0);
+        REQUIRE(dbInflationWinners.size() == 10);
+    }
+    REQUIRE(memoryInflationWinners == dbInflationWinners);
+    REQUIRE(mixedInflationWinners == dbInflationWinners);
+}
+
+// TODO(jonjove): Might be a few more cases worth testing here
+TEST_CASE("LedgerState::rollback", "[ledger][ledgerstate][rollback]")
+{
+    for (auto val : xdr::xdr_traits<LedgerEntryType>::enum_values())
+    {
+        LedgerEntryType let = static_cast<LedgerEntryType>(val);
+        std::string letStr = xdr::xdr_traits<LedgerEntryType>::enum_name(let);
+        SECTION(letStr)
+        {
+            VirtualClock clock;
+            Config cfg = getTestConfig();
+            cfg.INVARIANT_CHECKS = {};
+            Application::pointer app = createTestApplication(clock, cfg);
+
+            auto le1 = generateLedgerEntry(let);
+            auto le2 = generateLedgerEntryWithSameKey(le1);
+            auto key = LedgerEntryKey(le1);
+            REQUIRE(key == LedgerEntryKey(le2));
+
+            LedgerStateRoot& root(app->getLedgerStateRoot());
+            {
+                LedgerState ls(root);
+                ls.create(le1);
+                ls.rollback();
+            }
+            {
+                LedgerState ls(root);
+                REQUIRE_THROWS_WITH(ls.load(key),
+                                    "Key does not exist in database");
+            }
+
+            {
+                LedgerState ls(root);
+                ls.create(le1);
+                {
+                    LedgerState lsChild(ls);
+                    *lsChild.load(key)->entry() = le2;
+                    lsChild.rollback();
+                }
+                REQUIRE(*ls.load(key)->entry() == le1);
+            }
+        }
+    }
+}
+
+TEST_CASE("Exception during LedgerState::commit leaves cache valid",
+          "[ledger][ledgerstate][flushcache]")
+{
+    for (auto val : xdr::xdr_traits<LedgerEntryType>::enum_values())
+    {
+        LedgerEntryType let = static_cast<LedgerEntryType>(val);
+        std::string letStr = xdr::xdr_traits<LedgerEntryType>::enum_name(let);
+        SECTION(letStr)
+        {
+            VirtualClock clock;
+            Config cfg = getTestConfig();
+            cfg.INVARIANT_CHECKS = {};
+            Application::pointer app = createTestApplication(clock, cfg);
+
+            auto le = generateLedgerEntry(let);
+            auto key = LedgerEntryKey(le);
+
+            LedgerStateRoot& root(app->getLedgerStateRoot());
+            {
+                LedgerState ls(root);
+                ls.create(le);
+                ls.commit();
+            }
+            {
+                LedgerState ls(root);
+                ls.load(key)->erase();
+                REQUIRE_THROWS_AS(ls.commit([]() {
+                    throw std::runtime_error("Abort database commit");
+                }),
```

Comment:
did you forget clang-format or is it clang-format issue? This few lines of code looks strange